awx/main/tasks.py

```diff
@@ -429,46 +429,57 @@ def execution_node_health_check(node):
 
 
 def inspect_execution_nodes(instance_list):
-    node_lookup = {}
-    for inst in instance_list:
-        if inst.node_type == 'execution':
-            node_lookup[inst.hostname] = inst
+    with advisory_lock('inspect_execution_nodes_lock', wait=True):
+        node_lookup = {}
+        for inst in instance_list:
+            if inst.node_type == 'execution':
+                node_lookup[inst.hostname] = inst
+
+        ctl = get_receptor_ctl()
+        connections = ctl.simple_command('status')['Advertisements']
+        nowtime = now()
+        for ad in connections:
+            hostname = ad['NodeID']
+            commands = ad.get('WorkCommands') or []
+            if 'ansible-runner' not in commands:
+                continue
+            changed = False
+            if hostname in node_lookup:
+                instance = node_lookup[hostname]
+            else:
+                defaults = dict(enabled=False)
```

Comment:
ignoring the rest of the diff (which is due to the indentation change), this is where I actually made some tweaks:
* disable the instance by default
* .. and add the instance to the `default` IG by way of the `policy_instance_list` (after fishing around a bunch, I determined this seemed like the 'right' way to assign the instance to the group -- tell the IG what we want, and let our `apply_cluster_membership_policies()` task be responsible for making it so.)